conf/machine/raspberrypi-3-b-plus.conf

```diff
@@ -0,0 +1,63 @@
+# Raspberry Pi profile
+# Copyright (c) Cybertrust Japan Co., Ltd.
+#
+# Authors:
+#  Masami Ichikawa <masami.ichikawa@miraclelinux.com>
+#
+# SPDX-License-Identifier: MIT
+#
+
+DISTRO_ARCH ?= "arm64"
+
+# Raspberry Pi needs non-free-firmware repository.
+DISTRO_APT_SOURCES += "conf/distro/emlinux-bookworm-raspberrypi.list"
+
+# wic file settings
+IMAGE_FSTYPES ?= "wic wic.xz"
+WKS_FILE ?= "rpi-sdimg"
+
+IMAGE_PREINSTALL += "\
+  firmware-brcm80211 \
+"
+IMAGE_INSTALL += "\
+  u-boot-raspberrypi-3-b-plus \
+  rpi-u-boot-scr \
+  rpi-boot-config \
+  raspberrypi-bootfiles \
+"
+
+WIC_IMAGER_INSTALL = "\
+  parted \
+  dosfstools \
+  mtools \
+  e2fsprogs \
+  python3-distutils \
+  bmap-tools \
+"
+
+KERNEL_DEFCONFIG = "raspberrypi-3-b-plus_defconfig"
+
+PREFERRED_PROVIDER_u-boot-raspberrypi-3-b-plus = "u-boot-raspberrypi-3-b-plus"
+U_BOOT_CONFIG:raspberrypi-3-b-plus = "rpi_3_b_plus_defconfig"
+U_BOOT_BIN:raspberrypi-3-b-plus = "u-boot.bin"
+
+KERNEL_IMAGE = "Image"
+DTB_FILES = "bcm2837-rpi-3-b-plus.dtb"
+
+IMAGE_BOOT_FILES ?= "${KERNEL_IMAGE} ${DTB_FILES} \
+                     ${IMAGE_ROOTFS}/boot/overlays/*;overlays/ \
+                     ${IMAGE_ROOTFS}/usr/lib/u-boot/raspberrypi-3-b-plus/u-boot.bin;./ \
```

Comment:
Maybe ``` ${IMAGE_ROOTFS}/usr/lib/u-boot/${MACHINE}/u-boot.bin ``` would be  better when we need to support raspberry pi 4 or so in the future?

Reply:
We would have to add variables to or separate other places other than what you pointed to share the same code for supporting raspberry pi 4.

At least, the following firmware files are also specific to raspberrypi-3-b-plus, so just applying `MACHINE` to the u-boot path would be meaningless.
```
                     ${IMAGE_ROOTFS}/boot/start.elf;./ \
                     ${IMAGE_ROOTFS}/boot/start_cd.elf;./ \
                     ${IMAGE_ROOTFS}/boot/start_db.elf;./ \
                     ${IMAGE_ROOTFS}/boot/start_x.elf;./ \
                     ${IMAGE_ROOTFS}/boot/fixup.dat;./ \
                     ${IMAGE_ROOTFS}/boot/fixup_cd.dat;./ \
                     ${IMAGE_ROOTFS}/boot/fixup_db.dat;./ \
                     ${IMAGE_ROOTFS}/boot/fixup_x.dat;./"
```

Reply:
I have adopted this suggestion here.

https://github.com/miraclelinux/meta-emlinux/pull/241/files#diff-c7038d6a916b27413d069d7d655ca7213f9a6a08463865a9c86f1f5d4cca7d60R42

This .conf file is going to be shared by the Raspberry Pi 3 model series . The suggestion was useful to share the .conf file though I do not suppose the Raspberry Pi 4 support yet.

Reply:
okay